app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts

```diff
@@ -34,8 +57,9 @@ export function findExistingNetwork(chainId, findNetworkConfigurationBy) {
   return findNetworkConfigurationBy({ chainId });
 }
 
-export function validateChainId(chainId) {
-  const _chainId = typeof chainId === 'string' && chainId.toLowerCase();
+export function validateChainId(chainId: Hex): Hex {
+  const _chainId = chainId.toLowerCase() as Hex;
```

Comment:
@MajorLift When we do `chainId.toLowerCase()`, it seems to be returning type `string`. Is there any way for `chainId` to be  `Hex` without the `as` assertion?

Reply:
We can also safely define a type guard here, since any string that's validated by `isPrefixedFormattedHexString` must be a `Hex` type.

```diff
diff --git a/app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts b/app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts
index 002d6554fc..8622b1047c 100644
--- a/app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts
+++ b/app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts
@@ -58,9 +58,13 @@ export function findExistingNetwork(
 }
 
 export function validateChainId(chainId: Hex): Hex {
-  const _chainId = chainId.toLowerCase() as Hex;
+  const _chainId = chainId.toLowerCase();
 
-  if (!isPrefixedFormattedHexString(_chainId)) {
+  if (
+    !((value: string): value is Hex => isPrefixedFormattedHexString(value))(
+      _chainId,
+    )
+  ) {
     throw ethErrors.rpc.invalidParams({
       message: `Expected 0x-prefixed, unpadded, non-zero hexadecimal string 'chainId'. Received:\n${chainId}`,
     });

```

Reply:
`toLowerCase` does unnecessarily widen the type of `chainId`, and an as assertion is one way to resolve this. 

However, there's actually a better way to reason about the type of `_chainId`. Because the code immediately invokes the validator function `isPrefixedFormattedHexString`, we are provided with a runtime guarantee that any `_chainId` value that doesn't throw an error in the first if-block will be of a type narrower than `Hex`.

So now all we have to do is make sure this guarantee is reflected at the type-level as well, which we achieve by using the type guard with the `value is Hex` predicate.

This successfully narrows `_chainId` back to `Hex` past the first if-block. Without relying on any assertions!